weather_data_reader.py

```diff
@@ -0,0 +1,43 @@
+import calendar
+import os
+
+from parse_weather_data import WeatherParser
+from weather_calculation import WeatherCalculations
+
+
+class WeatherDataReader:
+    
+    def __init__(self, directory):
+        self.directory = directory
+        self.parser = WeatherParser()
+        self.weather_stats = WeatherCalculations()
+
+    def _get_monthly_file_path(self, year, month):
+        if isinstance(month, str):
+            month = list(calendar.month_abbr).index(month.capitalize())
```

Comment:
I first used try except block here. But then i saw that instance could not be a random string because first of all the argument accepted by user for year report should be in int. So if the user provides any random string as input the program will throw invalid int value error. Secondly the month names are generated by in built functions so it could not produce any random string that will break the flow.

Reply:
I read somewhere that methods that are used internally should start with leading underscores. I was under the impression that these are those methods that are not imported in other files and are used within the same file. But now i read that article in detail, i know that this is not the case and its related to being private and public.